deebot_client/commands/json/pads_cleaning_interval.py

```diff
@@ -0,0 +1,39 @@
+"""Pads cleaning interval commands."""
+from typing import Any
+
+from deebot_client.command import InitParam
+from deebot_client.event_bus import EventBus
+from deebot_client.events.pads_cleaning_interval import PadsCleaningIntervalEvent
+from deebot_client.message import HandlingResult, MessageBodyDataDict
+
+from .common import CommandWithMessageHandling, SetCommand
+
+
+class GetPadsCleaningInterval(CommandWithMessageHandling, MessageBodyDataDict):
+    """Get pads cleaning interval command."""
+
+    name = "getPadsCleaningInterval"
+
+    @classmethod
+    def _handle_body_data_dict(
+        cls, event_bus: EventBus, data: dict[str, Any]
+    ) -> HandlingResult:
+        """Handle message->body->data and notify the correct event subscribers.
+
+        :return: A message response
+        """
+        event_bus.notify(PadsCleaningIntervalEvent(int(data["interval"])))
+        return HandlingResult.success()
+
+
+class SetPadsCleaningInterval(SetCommand):
+    """Set pads cleaning interval command."""
+
+    name = "setPadsCleaningInterval"
```

Comment:
Is this command correct?
Is it related to your comment in [issue-1918161365](https://github.com/DeebotUniverse/Deebot-4-Home-Assistant/issues/434#issue-1918161365) in file [Ecovacs T20 Cleaning pads interval.txt](https://github.com/DeebotUniverse/Deebot-4-Home-Assistant/files/12753366/Ecovacs.T20.Cleaning.pads.interval.txt)?
In the file the command is `setWashInterval` but here used `setPadsCleaningInterval`

Reply:
It works!

I changed both the name in the set method and in the get method. Because I based my work on the work mode which had the name of the set function identical to the json cmdName, I didn't realize what this "name variable" was.

I still have a question though, for the get function, I put name = "getWashInterval" because it seemed to be the thing to do but I never saw a getWashInterval in a json, is it used? How one would know for sure what to put instead of simply guessing?

Reply:
nice :)

**get** and **set** are in all cases, I always saw, same named.
The **get** is always called when the app/screen with info is opened. 
When you not saw the **get** request directly, it could be it was not called in a single request, instead it was called in a multi **get** request, which you see in the **`getInfo`** request like this example:

```json
{
  "request": {
    "td": "q",
    "toId": "REPLACED",
    "toType": "p95mgv",
    "toRes": "REPLACED",
    "cmdName": "getInfo",
    "payloadType": "j",
    "auth": {
      "with": "REPLACED",
      "userid": "REPLACED",
      "realm": "REPLACED",
      "token": "REPLACED",
      "resource": "REPLACED"
    },
    "app": {
      "id": "ecovacs"
    },
    "payload": {
      "body": {
        "data": [
          "getWaterInfo",
          "getCleanPreference",
          "getStationInfo",
          "getAutoEmpty",
          "getSweepMode"
        ]
      },
      "header": {
        "pri": 2,
        "ts": "1699216144028",
        "tzm": 60,
        "ver": "0.0.22"
      }
    }
  },
  "response": {
    "id": "REPLACED",
    "ret": "ok",
    "resp": {
      "header": {
        "pri": 1,
        "tzm": 60,
        "ts": "1699216136655",
        "ver": "0.0.1",
        "fwVer": "1.7.2",
        "hwVer": "0.1.1",
        "wkVer": "0.1.54"
      },
      "body": {
        "code": 0,
        "msg": "ok",
        "data": {
          "getWaterInfo": {
            "code": 0,
            "msg": "ok",
            "data": {
              "enable": 0,
              "amount": 2,
              "type": 0,
              "sweepType": 1
            }
          },
          "getCleanPreference": {
            "data": {
              "enable": 1
            },
            "code": 0,
            "msg": "ok"
          },
          "getStationInfo": {
            "data": {
              "state": 1,
              "name": "Empty Station",
              "model": "AES",
              "sn": "317211735002292000387",
              "wkVer": "0.1.3"
            },
            "code": 0,
            "msg": "OK"
          },
          "getAutoEmpty": {
            "code": 0,
            "msg": "OK",
            "data": {
              "status": 0,
              "enable": 1,
              "frequency": "auto"
            }
          }
        }
      }
    },
    "payloadType": "j"
  }
}
```

Reply:
Ah yes thanks, I did see such json but didn't pay much attention because I didn't think they would be usefull. I'll add the one for T20 in change request #434